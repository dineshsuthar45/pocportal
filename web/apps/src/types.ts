import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
