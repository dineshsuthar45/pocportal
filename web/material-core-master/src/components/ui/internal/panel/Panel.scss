.panel-root {
	padding: 10px;
}
