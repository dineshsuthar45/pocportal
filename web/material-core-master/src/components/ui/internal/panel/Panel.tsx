import { Box } from "@mui/material";
import "./Panel.scss";
import { PanelProps } from "./Panel.types";

/**
 * Tab component
 */
export const MPanel = ({ children, value, index }: PanelProps) => (
  <div className="panel-root" hidden={value !== index} id={`${index}`}>
    {value === index && <Box>{children}</Box>}
  </div>
);

MPanel.defaultProps = {};
