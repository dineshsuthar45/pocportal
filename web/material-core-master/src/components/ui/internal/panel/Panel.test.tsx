import { render } from "@testing-library/react";
import React from "react";
import { PanelProps } from "./Panel.types";
import { MPanel } from "./Panel";

describe("Panel", () => {
  const renderComponent = () => render(<MPanel {...props} />);
  let props: PanelProps;

  beforeEach(() => {
    props = {
      children: <div>Test</div>,
      index: 1,
      value: 1,
    };
  });

  it("should render correctly", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
