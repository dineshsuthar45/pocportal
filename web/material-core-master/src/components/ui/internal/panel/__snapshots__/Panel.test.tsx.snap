// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should render correctly 1`] = `
<div>
  <div
    class="panel-root"
    id="1"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</div>
`;
